### ISSUE TITLE
FinalLocalVariable: false-negative with anonymous class

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#4727]
found at https://github.com/checkstyle/checkstyle/pull/4719#pullrequestreview-50190684

```
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
     <module name="TreeWalker">
        <module name="FinalLocalVariable"/>
    </module>
</module>

$ cat Test.java 
public class Test {
    public void test() {
        Object testSupport = new Object() {
            @Override
            public String toString() {
                final String dc = new String(); // comment this line to make violation happen
                return dc;
            }
        };
        testSupport.toString();
     }
}

$ java -jar checkstyle-8.0-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```

Expected:
```
[ERROR] Test.java:3:16: Variable 'testSupport' should be declared final. [FinalLocalVariable]
```

**Comment by rnveach** [comment]
Issue is https://github.com/checkstyle/checkstyle/blob/073c370de2e65fc346b51ef09763ecb30b5cd14f/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java#L213 .
`branchContains` is checking all children instead of the modifiers of the current variable.
It thinks code is written like `final Object testSupport` instead of seeing that the `final` is inside the inner anonymous class.

**Comment by romani** [comment]
fix is merged.
